Cannot build unit_tests on ARM: conflicting declaration for float types

Got things like
```
In file included from /home/parallels/code/antelope_repos/spring/libraries/boost/libs/unordered/include/boost/unordered/detail/foa/core.hpp:62,
                 from /home/parallels/code/antelope_repos/spring/libraries/boost/libs/unordered/include/boost/unordered/detail/foa/table.hpp:18,
                 from /home/parallels/code/antelope_repos/spring/libraries/boost/libs/unordered/include/boost/unordered/unordered_flat_map.hpp:14,
                 from /home/parallels/code/antelope_repos/spring/unittests/savanna_cluster.hpp:8,
                 from /home/parallels/code/antelope_repos/spring/unittests/forked_tests_savanna.cpp:1:
/usr/lib/gcc/aarch64-linux-gnu/11/include/arm_neon.h:72:16: error: conflicting declaration ‘typedef __fp16 float16_t’
   72 | typedef __fp16 float16_t;
      |                ^~~~~~~~~
In file included from /home/parallels/code/antelope_repos/spring/libraries/softfloat/source/include/softfloat.hpp:3,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/database_utils.hpp:5,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/code_object.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/wasm_interface.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/config.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/block_timestamp.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/block_header.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/block_header_state_legacy.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/block_state_legacy.hpp:3,
                 from /home/parallels/code/antelope_repos/spring/libraries/chain/include/eosio/chain/controller.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/libraries/testing/include/eosio/testing/tester.hpp:2,
                 from /home/parallels/code/antelope_repos/spring/unittests/savanna_cluster.hpp:6,
                 from /home/parallels/code/antelope_repos/spring/unittests/forked_tests_savanna.cpp:1:
/home/parallels/code/antelope_repos/spring/libraries/softfloat/source/include/softfloat_types.h:50:32: note: previous declaration as ‘typedef struct float16_t float16_t’
   50 | typedef struct { uint16_t v; } float16_t;
      |                                ^~~~~~~~~

```

And similar issue for some other types such as float32_t

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,6 +179,10 @@ set(CMAKE_DISABLE_FIND_PACKAGE_OpenSSL On)
 # is effectively --catch_system_errors=no broadly across all tests
 add_compile_definitions(BOOST_TEST_DEFAULTS_TO_CORE_DUMP)
 
+# Unless this is set, some boost unordered headers include `arm_neon.h` when building on ARM processors,
+# which includes a definition for `float16_t` conflicting with the one in `libraries/softfloat`.
+add_compile_definitions(BOOST_UNORDERED_DISABLE_NEON)
+
 add_subdirectory( libraries )
 add_subdirectory( plugins )
 add_subdirectory( programs )

--- a/unittests/savanna_cluster.hpp
+++ b/unittests/savanna_cluster.hpp
@@ -5,6 +5,8 @@
 
 #include <eosio/testing/tester.hpp>
 #include <ranges>
+
+#define BOOST_UNORDERED_DISABLE_NEON  // because of conflicting declaration for float16_t in softfloat (issue #517)
 #include <boost/unordered/unordered_flat_map.hpp>
 
 #include "snapshot_suites.hpp"

--- a/unittests/savanna_cluster.hpp
+++ b/unittests/savanna_cluster.hpp
@@ -6,7 +6,6 @@
 #include <eosio/testing/tester.hpp>
 #include <ranges>
 
-#define BOOST_UNORDERED_DISABLE_NEON  // because of conflicting declaration for float16_t in softfloat (issue #517)
 #include <boost/unordered/unordered_flat_map.hpp>
 
 #include "snapshot_suites.hpp"

